app/controllers/analysis_controller.rb

```diff
@@ -0,0 +1,420 @@
+# Analysis controller
+require 'json'
```

Comment:
I'm not sure you need this, I'm pretty sure it should already be pulled in by Rails.